lib/storage.js

```diff
@@ -53,6 +53,14 @@ async function getAsset(assetReference, directory, name, disableDownload) {
         const source = new Asset(assetReference, directory, name);
         if (disableDownload) {
             console.log(`Skipping source file download for ${AssetComputeLogUtils.redactUrl(source.url)}`);
+            const protocol = URL.parse(assetReference.url).protocol;
```

Comment:
I think we verify the URL validity before, but please double-check. In case it is not done already, please first verify the URL is valid using https://www.npmjs.com/package/valid-data-url (it is already used in other places of the SDK).

Reply:
Yes this is done as part of worker constructor
  ```
function validateParameters(params)
if (params.source !== undefined && params.source !== null) {
        if (typeof params.source === 'string') {
            params.source = { url: params.source };
        }

        if (!process.env.WORKER_TEST_MODE) {
            if (params.source.url.startsWith('data:')) {
                if (!validDataUrl(params.source.url)) {
                    throw new SourceCorruptError(`Invalid or missing data url ${params.source.url}`);
                }
            } else if ((!isHttpsUri(params.source.url))) {
                throw new SourceUnsupportedError(`Invalid or missing https url ${params.source.url}`);
            }
        }
    }
```